test/integration/app-document/pages/_document.js

```diff
@@ -32,27 +25,17 @@ export default class MyDocument extends Document {
   }
 
   render () {
-    let csp
-    switch (this.props.withCSP) {
-      case 'hash':
-        csp = `default-src 'self'; script-src 'self' ${cspHashOf(NextScript.getInlineScriptSource(this.props))}; style-src 'self' 'unsafe-inline'`
-        break
-      case 'nonce':
-        csp = `default-src 'self'; script-src 'self' 'nonce-test-nonce'; style-src 'self' 'unsafe-inline'`
-        break
-    }
-
     return (
       <html>
-        <Head nonce='test-nonce'>
-          {csp ? <meta httpEquiv='Content-Security-Policy' content={csp} /> : null}
+        <Head>
+          {this.props.withCSP ? <meta httpEquiv='Content-Security-Policy' content="default-src 'self'; style-src 'self' 'unsafe-inline';" /> : null}
```

Comment:
This is actually weakening the current CSP of the example. Why are you suggesting this? It does not make sense to me, users should rather define a hash in their CSP for the exceptional inline scripts they need, or nonces (if they run a server to reliably generate fresh nones, of course).

Reply:
Not true at all. Look closely at your CSP policy when you were using hashes: https://github.com/zeit/next.js/blob/canary/test/integration/app-document/pages/_document.js#L38

Inline styles is a different issue and a much more minor issue as it’s only a tiny visual security hole (instead of something being executed).

Reply:
You're right, I misread that rule.